README.md

```diff
@@ -103,13 +103,50 @@ data:
 
 #### Set up Cloud Backup With GCP Secret Manager
 
-The `secret-agent` expects credentials to be discoverable via standard [GCP mechanisms](https://cloud.google.com/docs/authentication). These credentials can be provided in a number of ways, including user accounts and service accounts.
+The `secret-agent` expects credentials to be discoverable via standard [GCP mechanisms](https://cloud.google.com/docs/authentication). These credentials can be provided in a number of ways, including:
+
+* [Workload identity](#workload-identity) (recommended for GKE deployments)
+* [Credentials file](#credentials-file) with user accounts or service accounts.
 
 Please refer to the [GCP Documentation](https://cloud.google.com/secret-manager/docs/reference/libraries?hl=nl#cloud-console) for instructions on how to create a service account with the necessary permissions to access the GCP Secrets Manager. The `secret-agent` needs access to read/write secrets. This can be achieved by assigning the `Secret Manager Admin` role to the service account provided.
 
+##### Workload identity
+
+Workload Identity is the recommended way to access Google Cloud services from applications running within GKE. For more information on how to enable workload identity see [GCP Documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#gcloud_1).
+
+In general, once the Google service account has been created with the proper role attached and workload identity has been enabled for your GKE cluster and nodes, you can run the following commands to enable workload identity for the `secret-agent`:
+
+```bash
+PROJECTID=myproject #GCP project ID
+GSA-NAME=mygcpserviceaccount #GCP service account name
+# Create the GCP IAM policy binding
+gcloud iam service-accounts add-iam-policy-binding --role roles/iam.workloadIdentityUser --member "serviceAccount:${PROJECTID}.svc.id.goog[secret-agent-system/secret-agent-manager-service-account]" ${GSA-NAME}@${PROJECTID}.iam.gserviceaccount.com
+# Annotate the Kubernetes service account
+kubectl -n secret-agent-system annotate serviceaccounts secret-agent-manager-service-account iam.gke.io/gcp-service-account=${GSA-NAME}@${PROJECTID}.iam.gserviceaccount.com
+```
+
+**Note: in order to use workload identity, no `spec.appConfig.credentialsSecretName` should be provided**. If credentials are provided, `secret-agent` will use the provided credentials instead.
```

Comment:
I'm wondering if we should switch around Azure logic to behave this way too....

Reply:
I do fear that without an explicitly set flag for "use workload id" users could run into and issue where they expect it to work but it doesn't.

Reply:
Not sure about this. Take AWS as an example, we don't have a setting to use instance metadata or not. The workload identify setup happens outside the operator itself so there's nothing we do to turn it on or off.

Reply:
updated README.md to clarify azure's "managed identity" is used if no secret is provided